tests/tests/tier0/bluechi-is-online-agent-wait/test_bluechi_is_online_agent_wait.py

```diff
@@ -0,0 +1,153 @@
+from bluechi_test.config import BluechiAgentConfig, BluechiControllerConfig
+from bluechi_test.machine import BluechiAgentMachine, BluechiControllerMachine
+from bluechi_test.test import BluechiTest
+
+LOGGER = logging.getLogger(__name__)
+
+NODE_FOO = "node-foo"
+
+
+class ResultFuture:
+    def __init__(self):
+        self.result = None
+        self.output = ""
+
+
+def check_agent(
+    bluechi_is_online: BluechiIsOnline,
+    future: ResultFuture,
+    description: str,
+    wait_time: int = 0,
+    expected_result: int = 0,
+):
+    result, output = bluechi_is_online.run(
+        description,
+        f"agent --wait {wait_time}",
+        check_result=False,
+        expected_result=expected_result,
+    )
+    future.result = result
+    future.output = output
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    node_foo = nodes[NODE_FOO]
+
+    # Test 1: Start agent and verify --wait option returns 0 immediately
+    node_foo.systemctl.start_unit("bluechi-agent")
+    assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "active")
+    LOGGER.debug("Starting test number 1 - agent should be online.")
+
+    result_future_online = ResultFuture()
+    check_thread_online = threading.Thread(
+        target=check_agent,
+        args=(
+            node_foo.bluechi_is_online,
+            result_future_online,
+            "Checking if agent is online",
+            0,
+            0,
+        ),
+    )
+    check_thread_online.start()
+    check_thread_online.join()
+    assert result_future_online.result == 0, "Expected agent to be online"
+
+    # Test 2: Stop agent and verify --wait returns 1 after wait time is expired.
+    node_foo.systemctl.stop_unit("bluechi-agent")
+    assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "inactive")
+    LOGGER.debug("Starting test number 2 - agent should remain offline.")
+
+    result_future_offline = ResultFuture()
+    check_thread_offline = threading.Thread(
+        target=check_agent,
+        args=(
+            node_foo.bluechi_is_online,
+            result_future_offline,
+            "Checking if agent remains offline",
+            5000,
+            1,
+        ),
+    )
+    check_thread_offline.start()
+    check_thread_offline.join()
+    assert result_future_offline.result != 0, "Expected agent to remain offline"
+
+    # Test 3: Stop agent, run 'bluechi-is-online agent --wait', start agent and verify --wait returns 0 before the
+    # wait time expires
+    LOGGER.debug(
+        "Starting test number 3, stopping agent to ensure it is inactive before starting `bluechi-is-online` "
+        "thread."
+    )
+    node_foo.systemctl.stop_unit("bluechi-agent")
+    assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "inactive")
+    LOGGER.debug("Agent confirmed inactive after stopping.")
+
+    # Add a short delay to ensure the "inactive" state is fully processed
+    time.sleep(1)
```

Comment:
This could be an issue with slower boards, can't we use `bluechictl` to wait for inactive state?